src/storage/store/column_chunk_data.cpp

```diff
@@ -388,14 +390,17 @@ void ColumnChunkData::setToInMemory() {
     }
 }
 
-void ColumnChunkData::resize(uint64_t newCapacity) {
+void ColumnChunkData::resize(uint64_t newCapacity, bool reserveDataAndInitializeToZero) {
     if (newCapacity > capacity) {
         capacity = newCapacity;
     }
     const auto numBytesAfterResize = getBufferSize(newCapacity);
     if (numBytesAfterResize > getBufferSize()) {
-        auto resizedBuffer = buffer->getMemoryManager()->mallocBuffer(true, numBytesAfterResize);
-        memcpy(resizedBuffer->getBuffer().data(), buffer->getBuffer().data(), getBufferSize());
+        auto resizedBuffer = buffer->getMemoryManager()->mallocBuffer(
+            reserveDataAndInitializeToZero, numBytesAfterResize);
```

Comment:
If we're going to overwrite the data afterwards, it might make more sense to not initialize the buffer to zero, overwrite that data and then zero just the part exceeding the original buffer's size. Otherwise we're initializing the part that already exists twice.